### ISSUE TITLE
How long should I train the model?

### ISSUE THREAD

**Issue by Purnay04** [hoangminhtoan/a2m_osnet_keras#2]
I have finished my code section. So I started training with `28841` training images and `15905` validation images shards into `32` training and `32` validation files.

**my command for training model is**
```python train.py --dropout_rate 0.2  --dataset_dir ./Market-1501-v15.09.15/shards --num_classes 1501 --optimizer adam 
--batch_size 16 --lr_sched linear --label_smoothing --initial_lr 1e-3 --final_lr 1e-4 --weight_decay 1e-4 --epochs 50 
--model osnet --num_data_workers 8
```
_but,_
model is getting under-fit with Augmentation and without Augmentation also. These are the results of training.....

![training_with_aug](https://user-images.githubusercontent.com/35885285/108324450-ee0dc600-71ed-11eb-8319-d303eeaf4632.png)

![training_without_aug](https://user-images.githubusercontent.com/35885285/108324463-f108b680-71ed-11eb-87f8-a7c2e77db404.png)

So, what should I set the value of `epochs` and `batches` or is there anything is wrong with my code or my data?

**Comment by sakura811** [comment]
> I have finished my code section. So I started training with `28841` training images and `15905` validation images shards into `32` training and `32` validation files.
> 
> **my command for training model is**
> 
> ```
> python train.py --dropout_rate 0.2  --dataset_dir ./Market-1501-v15.09.15/shards --num_classes 1501 --optimizer adam 
> --batch_size 16 --lr_sched linear --label_smoothing --initial_lr 1e-3 --final_lr 1e-4 --weight_decay 1e-4 --epochs 50 
> --model osnet --num_data_workers 8
> ```
> 
> _but,_
> model is getting under-fit with Augmentation and without Augmentation also. These are the results of training.....
> 
> ![training_with_aug](https://user-images.githubusercontent.com/35885285/108324450-ee0dc600-71ed-11eb-8319-d303eeaf4632.png)
> 
> ![training_without_aug](https://user-images.githubusercontent.com/35885285/108324463-f108b680-71ed-11eb-87f8-a7c2e77db404.png)
> 
> So, what should I set the value of `epochs` and `batches` or is there anything is wrong with my code or my data?

Hi, did you try the paremeters which mentioned in osnet paper, although that is for pytorch version.
Btw, could you share your tensorflow version osnet ?

**Comment by Purnay04** [comment]
Hey @sakura811 Which parameters are you talking about? for OSNet, I'm using `tensorflow-gpu = 2.3.0`.  I pushed my code. Check it out using this [link](https://github.com/Purnay04/a2m_osnet_keras).

and Hey @hoangminhtoan, my model is finally improving some accuracy but still the loss is directly proportional with accuracy and as I'm training this model using linear learning schedule so that the model stuck in between `50-57 accuracy.`
Using this 57% accuracy model, model still not able to predict. Can someone help me to get tuned this model.

**Comment by sakura811** [comment]
> Hey @sakura811 Which parameters are you talking about? for OSNet, I'm using `tensorflow-gpu = 2.3.0`. I pushed my code. Check it out using this [link](https://github.com/Purnay04/a2m_osnet_keras).
> 
> and Hey @hoangminhtoan, my model is finally improving some accuracy but still the loss is directly proportional with accuracy and as I'm training this model using linear learning schedule so that the model stuck in between `50-57 accuracy.`
> Using this 57% accuracy model, model still not able to predict. Can someone help me to get tuned this model.

@Purnay04 Hi, I mean that the OSNet paper mentioned the epoch number and batch size, but they are for pytorch version. I am not sure if it is useful for you.
paper: Batch size and weight decay are set to 64 and 5e-4 respectively. For training from scratch, SGD is used to train the network for 350 epochs. The learning rate starts from 0.065 and is decayed by 0.1 at 150, 225 and 300 epochs. Data augmentation includes random flip, random crop and random patch5.
And thanks for your link~

**Comment by Purnay04** [comment]
Thank you @sakura811. thank you for information. I will be setting up my model for training with your parameter and data augmentation. afterwards I'll post here the mAP and rank1 metrics.
